src/core_codemods/semgrep/semgrep_nan_injection.py

```diff
@@ -63,6 +60,15 @@ def replace_with_if_else(
         new_statement = cst.parse_statement(code)
         return new_statement.with_changes(leading_lines=updated_node.leading_lines)
 
+    def _get_var_in_call(self, node: cst):
+        match node.args[0].value:
+            case cst.Name():
+                # float(var)
+                return node.args[0].value.value
+            case cst.BinaryOperation():
+                # bool(float(var)), complex(float(var)), etc
+                return node.args[0].value.left.args[0].value.value
+
```

Comment:
This looks weird.
The type hint for `node` should be `cst.Call`.
The case for `cst.BinaryOperation()` does not match the the intention, at least judging from the comment above. Do you mean `cst.Call`? 
If that's the case, you should account for deeper nestings (e.g. `float(complex(int(...)))` ). Does the semgrep rule always guarantee a variable name there? It could be an expression.

Reply:
whoops, updated the type hint.
BinaryOperation because when nested cst really does interpret it as a binaryoperation, for all calls of bool, int, float, and complex. I checked them all.
```all(
    func=Name(
        value='bool',
        lpar=[],
        rpar=[],
    ),
    args=[
        Arg(
            value=BinaryOperation(
                left=Call(
                    func=Name(
                        value='float',
                        lpar=[],
                        rpar=[],
                    ),
                    args=[
                        Arg(
                            value=Name(
                                value='tid',
                                lpar=[],
                                rpar=[],
                            ),

```
I'll check the last part you mention.

Reply:
so checking in the [playground](https://semgrep.dev/playground/r/python.django.security.nan-injection.nan-injection?editorMode=advanced)
1. looks like infinite nesting is a possibility ... `complex(bool(float(tid))` will match. But this only matches as long as all funcs are one of complex, bool, float. Note the  `pattern-sanitizers`, that means any other function will sanitize this away.
2. I do need to handle the non-var case for `float(request.POST.get("tid"))`, which is annoying it  wasn't a semgrep example.

I think the reasonable thing to do here is to not implement #1, just test and not to it, but to do #2. Any other pattern won't be fixed for now.